pkg/routeagent_driver/handlers/ovn/connection.go

```diff
@@ -186,45 +188,85 @@ func getTLSConfig(k8sClientset clientset.Interface) (*tls.Config, error) {
 	return tlsConfig, nil
 }
 
-func discoverOvnKubernetesNetwork(ctx context.Context, k8sClientset clientset.Interface, zoneName string) (string, error) {
-	ovnDBPod, err := FindPod(ctx, k8sClientset, "name=ovnkube-db")
+func discoverOvnKubernetesNetwork(ctx context.Context, k8sClientset clientset.Interface,
+	dynamicClient dynamic.Interface, zoneName string,
+) (string, error) {
+	openshiftNetwork, err := FindOpenshiftNetwork(ctx, dynamicClient)
 	if err != nil {
 		return "", err
 	}
 
-	var nbdbAddress string
+	if openshiftNetwork != nil {
+		return discoverOpenshiftOvnKubernetesNetwork(ctx, k8sClientset)
+	}
+
+	return discoverKindOvnKubernetesNetwork(ctx, k8sClientset, zoneName)
+}
+
+/*
+
+openshift non-ic: in this case there will get the "app=ovnkube-node" and check for ovnkube-db service,
+if present we will the default Openshift db path
+
+openshift ic with one node per zone: if above service is not present we will return default Openshift socket path
+*/
+
+func discoverOpenshiftOvnKubernetesNetwork(ctx context.Context, k8sClientSet clientset.Interface) (string, error) {
+	ovnPod, err := FindPod(ctx, k8sClientSet, "app=ovnkube-node")
+	if err != nil {
+		return "", errors.Wrap(err, "error finding a pod with label \"app=ovnkube-node\"")
+	}
+
+	_, err = k8sClientSet.CoreV1().Services(ovnPod.Namespace).Get(ctx, ovnKubeService, metav1.GetOptions{})
+	if err == nil {
+		return defaultOpenshiftOVNNBDB, nil
```

Comment:
According to code it seems that there are only two options 
* non-ic ==> when both pod and service exist
* rest of the cases 
Is it correct?

 if the answer is yes , maybe it would be better to have a separate boolean function for detecting non-ic deployment.

Reply:
non-ic deployment identification is not same for kind and Openshift. So cannot have common for both.